Description: trigger layout after resizing the FrameView
 There are problems with QGLWidget based web plugins due to the fact
 that WebKit forces us to relayout during paint events, which in turn
 might resize widgets, causing the shared backing store's QImages to
 be corrupted and hence causing crashes.
 .
 This patch triggers a layout upon resizing the FrameView, which reduces
 the likelihood of resizing plugins during a subsequent paintEvent.
Origin: upstream, commits:
 https://github.com/annulen/webkit/commit/6faf11215e1af27d
 https://github.com/annulen/webkit/commit/76420459a13d9440
Bug: https://github.com/annulen/webkit/issues/511
Last-Update: 2019-01-19

--- a/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp
@@ -961,6 +961,8 @@
     FrameView* view = frame->view();
     ASSERT(view);
     view->resize(size);
+    if (view->needsLayout())
+        view->layout();
     view->adjustViewSize();
 }
 
